### DISCUSSION TITLE
postfixWithEndpointName() does not work in external converter

### DISCUSSION THREAD

**Discussion by grafalex82** [Koenkk/zigbee2mqtt#8732] [Devices]
My device implements several genOnOff endpoints. I am trying to use default fromZigbee/toZigbee on_off converters, but they do not work for me. 

Here is how I added it
```
const device = {
    zigbeeModel: ['Hello Zigbee Switch'],
    model: 'Hello Zigbee Switch',
    vendor: 'NXP',
    description: 'Hello Zigbee Switch',
    fromZigbee: [my_on_off],
    toZigbee: [tz.on_off],
    exposes: genEndpoints(1),
    endpoint: (device) => {
        return {button_1: 2}; // yes, endpoint numbering is correct for my device
    },
};

module.exports = device;
```

on_off code from fromZigbee.js

```

     on_off: {
        cluster: 'genOnOff',
        type: ['attributeReport', 'readResponse'],
        convert: (model, msg, publish, options, meta) => {
            if (msg.data.hasOwnProperty('onOff')) {
                const property = postfixWithEndpointName('state', msg, model);
                return {[property]: msg.data['onOff'] === 1 ? 'ON' : 'OFF'};
            }
        },
    },
```

I debugged this a little bit, and it looks like postfixWithEndpointName() calculate property name as 'state', instead of 'state_button_1' (my end point name). I took a look at postfixWithEndpointName() internals and it seems it is looking for 'meta' attribute of the 'model' object, but the model it gets does not have meta attribute. 

As a workaround I copied on_off implementation into my external converter code and calculate property name as 'state_button_1', which solved the issue. But it does not look correct from code reuse perspective.

What is the right way of adding multiple endpoints so that postfixWithEndpointName() works correctly?

**Comment by sjorge** [comment]
You need to set `multiEndpoint: true` also IIRC

**Reply by grafalex82** [reply]
I do not really get your answer. Could you please clarify?
1) Where should I enable multiEndpoint?
2) What is IIRC?

**Reply by sjorge** [reply]
As per the [readme](https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/README.md) for zhc

> - `multiEndpoint`: enables the multi endpoint functionallity in e.g. fromZigbee.on_off, example: normally this converter would return {"state": "OFF"}, when multiEndpoint is enabled the 'endpoint' method of the device definition will be called to determine the endpoint name which is then used as key e.g. {"state_left": "OFF"}. Only needed when device sends the same attribute from multiple endpoints. (default: false)


Example of a device were it is used:
https://github.com/Koenkk/zigbee-herdsman-converters/blob/952dd663dcdab8ef146995d77a2330ed13ccad47/devices/nodon.js#L26-L29

**Reply by grafalex82** [reply]
Thank you, adding 'meta' field to the device description helped. Thank you for your help.

From the ZHC documentation it was not really clear that 'meta' should be a part of the device description :(
